packages/contractkit/src/wallets/azure-hsm-wallet.test.ts

```diff
@@ -0,0 +1,325 @@
+      { name: 'chainId', type: 'uint256' },
+      { name: 'verifyingContract', type: 'address' },
+    ],
+    Person: [
+      { name: 'name', type: 'string' },
+      { name: 'wallet', type: 'address' },
+    ],
+    Mail: [
+      { name: 'from', type: 'Person' },
+      { name: 'to', type: 'Person' },
+      { name: 'contents', type: 'string' },
+    ],
+  },
+  primaryType: 'Mail',
+  domain: {
+    name: 'Ether Mail',
+    version: '1',
+    chainId: 1,
+    verifyingContract: '0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC',
+  },
+  message: {
+    from: {
+      name: 'Cow',
+      wallet: '0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826',
+    },
+    to: {
+      name: 'Bob',
+      wallet: '0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB',
+    },
+    contents: 'Hello, Bob!',
+  },
+}
+
+describe('AzureHSMWallet class', () => {
+  let wallet: AzureHSMWallet
+
+  // validate env file
+  beforeEach(() => {
+    // Use mock client if env vars not specified
+    if (!USING_MOCK) {
+      // Ensure all env vars are specified
+      expect(process.env.AZURE_CLIENT_ID).toBeDefined()
+      expect(process.env.AZURE_CLIENT_SECRET).toBeDefined()
+      expect(process.env.AZURE_TENANT_ID).toBeDefined()
+      expect(process.env.AZURE_VAULT_NAME).toBeDefined()
+      expect(process.env.AZURE_KEY_NAME).toBeDefined()
+    }
+
+    wallet = new AzureHSMWallet(AZURE_VAULT_NAME!)
+
+    if (USING_MOCK) {
+      jest
+        .spyOn<any, any>(wallet, 'generateNewKeyVaultClient')
+        .mockImplementation((_transport: any) => {
+          return {
+            getKeys: async (): Promise<string[]> => {
+              return Array.from(keyVaultAddresses.keys())
+            },
+            getPublicKey: async (keyName: string): Promise<BigNumber> => {
+              if (!keyVaultAddresses.has(keyName)) {
+                throw new Error(`Key ${keyName} not found in KeyVault ${AZURE_VAULT_NAME}`)
+              }
+              const privKey = keyVaultAddresses.get(keyName)!.privateKey
+              const pubKey = ethUtil.privateToPublic(ethUtil.toBuffer(privKey))
+              return new BigNumber(ensureLeading0x(pubKey.toString('hex')))
+            },
+            signMessage: async (message: Buffer, keyName: string): Promise<Signature> => {
+              if (keyVaultAddresses.has(keyName)) {
+                const trimmedKey = trimLeading0x(keyVaultAddresses.get(keyName)!.privateKey)
+                const pkBuffer = Buffer.from(trimmedKey, 'hex')
+                const signature = ethUtil.ecsign(message, pkBuffer)
+                return new Signature(signature.v - 27, signature.r, signature.s)
+              }
+              throw new Error(`Unable to locate key: ${keyName}`)
+            },
+          }
+        })
+    }
+  })
+
+  describe('without initializing', () => {
+    const knownAddress = ACCOUNT_ADDRESS1
+    let celoTransaction: Tx
+    beforeEach(() => {
+      return new Promise(async (resolve) => {
```

Comment:
It's equivalent to an async beforeEach that I was using when trying to debug some test issues. I've consolidated to the async approach since it's cleaner 👍